css/cssom-view/scroll-behavior-006.html

```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
+<title>Testing scrollOptions' behavior with scrollIntoView for nested scrolling nodes</title>
+<meta name="timeout" content="long"/>
+<link rel="author" title="Frédéric Wang" href="mailto:fwang@igalia.com">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#propdef-scroll-behavior">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#scrolling-box">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="support/scroll-behavior.js"></script>
+<style>
+  .scrollable {
+    overflow: auto;
+    height: 200px;
+  }
+  .smoothBehavior {
+    scroll-behavior: smooth;
+  }
+  .gradient {
+    background: linear-gradient(135deg, red, green);
+  }
+</style>
+<div id="log">
+</div>
+<div>
+  <div class="scrollable smoothBehavior" style="width: 450px">
+    <div class="gradient" style="width: 100px; height: 500px;"></div>
+    <div class="scrollable smoothBehavior" style="width: 400px">
+      <div class="gradient" style="width: 100px; height: 500px;"></div>
+      <div class="scrollable" style="width: 350px">
+        <div class="gradient" style="width: 100px; height: 500px;"></div>
+        <div class="scrollable" style="width: 300px">
+          <div class="gradient" style="width: 100px; height: 500px;"></div>
+          <div class="scrollable smoothBehavior" style="width: 250px">
+            <div class="gradient" style="width: 100px; height: 500px;"></div>
+            <div class="scrollable" style="width: 200px">
+              <div class="gradient" style="width: 100px; height: 500px;"></div>
+              <div id="elementToReveal" style="width: 10px; height: 10px; background: black;"></div>
+              <div class="gradient" style="width: 100px; height: 500px;"></div>
+            </div>
+            <div class="gradient" style="width: 100px; height: 500px;"></div>
+          </div>
+          <div class="gradient" style="width: 100px; height: 500px;"></div>
+        </div>
+      <div class="gradient" style="width: 100px; height: 500px;"></div>
+      </div>
+      <div class="gradient" style="width: 100px; height: 500px;"></div>
+    </div>
+  </div>
+</div>
+<script>
+  promise_test(() => {
+    // CSSOM seems to suggest revealing happens from innermost to outermost but some browsers do the opposite.
```

Comment:
Yeah, I think the issue with innermost to outermost is that the innermost scroller might be fully invisible to start, so you if you animate that scroller first, the user sees none of that animation. If you do outer-to-inner, each step is visible. So the test approach below seems the right way to go until there's more consistency in implementations.

Reply:
Actually, the spec, Gecko and Chromium have different approaches. I opened https://github.com/w3c/csswg-drafts/issues/3127 for now.

I can probably make the test slightly stricter by observing scroll positions and verifying each coordinate is non-decreasing, but probably not much more (in particular not checking smooth VS instant behaviors).